app/scripts/modules/netflix/fastProperties/view/history/FastPropertyHistory.tsx

```diff
@@ -0,0 +1,118 @@
+import * as React from 'react';
+import autoBindMethods from 'class-autobind-decorator';
+
+import { timestamp } from 'core/utils/timeFormatters';
+import { Property } from '../../domain/property.domain';
+import { fastPropertyReader } from '../../fastProperty.read.service';
+import { IPropertyHistoryEntry } from '../../domain/propertyHistory.domain';
+import { DiffView } from 'core/pipeline/config/actions/history/DiffView';
+
+import './FastPropertyHistory.less';
+
+interface IProps {
+  property: Property;
+}
+
+interface IState {
+  loading: boolean;
+  loadError: boolean;
+  entries: IPropertyHistoryEntry[];
+  diffsExpanded: number[];
+}
+
+@autoBindMethods
+export class FastPropertyHistory extends React.Component<IProps, IState> {
+
+  constructor(props: IProps) {
+    super(props);
+    this.state = {
+      loading: true,
+      loadError: false,
+      entries: [],
+      diffsExpanded: [],
+    };
+  }
+
+  public componentDidMount(): void {
+    this.loadHistory();
+  }
+
+  public componentWillReceiveProps(): void {
+    this.loadHistory();
+  }
+
+  private loadHistory(): void {
+    this.setState({loading: true});
+    const property = this.props.property;
+    fastPropertyReader.getHistory(property.propertyId, property.env)
+      .then(entries => this.setState({entries, loading: false, loadError: false}))
+      .catch(() => this.setState({loading: false, loadError: true}));
+  }
+
+  private showDiff(index: number) {
+    this.setState({diffsExpanded: this.state.diffsExpanded.concat(index)})
+  }
+
+  private hideDiff(index: number) {
+    this.setState({diffsExpanded: this.state.diffsExpanded.filter(i => i !== index)});
+  }
+
+  public render() {
+    const rows = [] as JSX.Element[];
+    this.state.entries.forEach((entry, index) => {
+      const diffExpanded = this.state.diffsExpanded.includes(index);
+      rows.push((
+        <tr key={index}>
+          <td>{timestamp(entry.timestamp)}</td>
+          <td>{entry.comment}</td>
+          <td>{entry.sourceOfUpdate}</td>
+          <td>{entry.updatedBy}</td>
+          <td>{entry.cmc}</td>
+          <td>
+            {!entry.diff && <span>n/a</span>}
+            {entry.diff && !diffExpanded && <a className="clickable" onClick={this.showDiff.bind(this, index)}>{'Show Diff'}</a>}
```

Comment:
with @autoBindMethods, do you still need the `.bind(this,...)` on these callbacks?